controllers/serviceinstance_controller.go

```diff
@@ -249,10 +313,74 @@ func (r *ServiceInstanceReconciler) createInstance(ctx context.Context, smClient
 	}
 
 	serviceInstance.Status.Ready = metav1.ConditionTrue
+
+	if serviceInstance.Spec.Shared == nil || (serviceInstance.Spec.Shared != nil && !*serviceInstance.Spec.Shared) {
+		serviceInstance.Status.Shared = metav1.ConditionFalse
+	}
+
 	setSuccessConditions(smClientTypes.CREATE, serviceInstance)
+
 	return ctrl.Result{}, r.updateStatus(ctx, serviceInstance)
 }
 
+func setConditionSharingNeedsToBeDone(object api.SAPBTPResource) {
+	conditions := object.GetConditions()
+
+	shouldShareCondition := metav1.Condition{
+		Type:               api.ConditionSharing,
+		Status:             metav1.ConditionFalse,
+		Reason:             getConditionReason(Updated, smClientTypes.PENDING),
+		Message:            "Sharing of instance needs to be performed",
+		ObservedGeneration: object.GetGeneration(),
+	}
+	meta.RemoveStatusCondition(&conditions, api.ConditionSharing)
+	meta.SetStatusCondition(&conditions, shouldShareCondition)
+
+	object.SetConditions(conditions)
+}
+
+func setConditionForFailedSharing(object api.SAPBTPResource, currentSharedStatus metav1.ConditionStatus, err error) {
```

Comment:
looks like duplicate code of setConditionForSuccessShareChange

Reply:
took out the common code